### PR TITLE
Add Descriptions for nodes in Builtinfunction

### PR DESCRIPTION
**Pull request by zora-wang**
@sharadkjaiswal . 
Nodes in builtinfunction have been added descriptions.

And all descriptions have been added in resources file

### CODE REVIEW

File: src/Engine/ProtoCore/Properties/Resources.en-US.resx [hunk 1]
```diff
@@ -384,4 +384,103 @@
   <data name="kStringIndexingCannotBeAssigned" xml:space="preserve">
     <value>String indexing cannot be assigned to, it is read only.</value>
   </data>
+  <data name="ChecksArrayWithUniformDepth" xml:space="preserve">
+    <value>Checks if the array has a uniform depth</value>
+  </data>
+  <data name="ChecksIfArrayContainsTheMember" xml:space="preserve">
+    <value>Checks if the array contains the member var</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.en-US.resx]: Checks if the array contains the given member

File: src/Engine/ProtoCore/Properties/Resources.en-US.resx [hunk 2]
```diff
@@ -384,4 +384,103 @@
...
+  </data>
+  <data name="ChecksIfTheArrayIsAllFalse" xml:space="preserve">
+    <value>Check if the array is all false</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.en-US.resx]: Checks if all elements in the specified array is false

**Comment by zora-wang** [comment]
@sharadkjaiswal , Resources file of builtinfunction is updated

File: src/Engine/ProtoCore/Lang/BuiltInMethods.cs [hunk 3]
```diff
@@ -464,7 +482,8 @@ public BuiltInMethods(Core core)
                         new KeyValuePair<string, ProtoCore.Type>("array", TypeSystem.BuildPrimitiveTypeObject(PrimitiveType.kTypeVar, Constants.kArbitraryRank)),
                         new KeyValuePair<string, ProtoCore.Type>("member", TypeSystem.BuildPrimitiveTypeObject(PrimitiveType.kTypeVar, Constants.kArbitraryRank))
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Lang/BuiltInMethods.cs]: Rename the parameter name "member" to "element".

File: src/Engine/ProtoCore/Lang/BuiltInMethods.cs [hunk 4]
```diff
@@ -475,7 +494,8 @@ public BuiltInMethods(Core core)
                         new KeyValuePair<string, ProtoCore.Type>("array", TypeSystem.BuildPrimitiveTypeObject(PrimitiveType.kTypeVar, Constants.kArbitraryRank)),
                         new KeyValuePair<string, ProtoCore.Type>("member", TypeSystem.BuildPrimitiveTypeObject(PrimitiveType.kTypeVar, 0))
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Lang/BuiltInMethods.cs]: Rename the parameter name "member" to "element", and the description string to ChecksIfArrayContainsTheElement

File: src/Engine/ProtoCore/Lang/BuiltInMethods.cs [hunk 5]
```diff
@@ -498,7 +518,8 @@ public BuiltInMethods(Core core)
                         new KeyValuePair<string, ProtoCore.Type>("array", TypeSystem.BuildPrimitiveTypeObject(PrimitiveType.kTypeVar, Constants.kArbitraryRank)),
                         new KeyValuePair<string, ProtoCore.Type>("member", TypeSystem.BuildPrimitiveTypeObject(PrimitiveType.kTypeVar, Constants.kArbitraryRank))
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Lang/BuiltInMethods.cs]: Rename the parameter name "member" to "element".

File: src/Engine/ProtoCore/Lang/BuiltInMethods.cs [hunk 6]
```diff
@@ -168,7 +169,9 @@ public BuiltInMethods(Core core)
                         new KeyValuePair<string, ProtoCore.Type>("array", TypeSystem.BuildPrimitiveTypeObject(PrimitiveType.kTypeVar, Constants.kArbitraryRank))
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Lang/BuiltInMethods.cs]: Rename the parameter name "array" to "list" at all the places.

File: src/Engine/ProtoCore/Lang/BuiltInMethods.cs [hunk 7]
```diff
@@ -211,6 +216,7 @@ public BuiltInMethods(Core core)
                         new KeyValuePair<string, ProtoCore.Type>("array2", TypeSystem.BuildPrimitiveTypeObject(PrimitiveType.kTypeVar, Constants.kArbitraryRank))
                     },
                     ID = BuiltInMethods.MethodID.kConcat,
+                    MethodAttributes = new MethodAttributes(){Description  = Resources.ReturnsConcatenatingArray}
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Lang/BuiltInMethods.cs]: This is same as List.Join but it fails in certain cases, whereas List.Join works better. Let's hide this method, by passing isHiddenInLibrary as true. 
Update the documentation to "Joins two lists into a single list."

File: src/Engine/ProtoCore/Properties/Resources.en-US.resx [hunk 8]
```diff
@@ -384,4 +384,103 @@
   <data name="kStringIndexingCannotBeAssigned" xml:space="preserve">
     <value>String indexing cannot be assigned to, it is read only.</value>
   </data>
+  <data name="ChecksArrayWithUniformDepth" xml:space="preserve">
+    <value>Checks if the array has a uniform depth</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.en-US.resx]: Let's replace array with list everywhere in the documentation.

**Comment by zora-wang** [comment]
@sharadkjaiswal , @elayabharath .
The descriptions for nodes in Builtin function are updated.
I have changed all the "array" parameters to "list"
and correct the grammar problems.

**Comment by zora-wang** [comment]
I deleted the original en-US file, and copy the Resources.resx and paste it to Resources.en-US.resx in order to make sure the two files are the same.

**Pull request merged**